### DISCUSSION TITLE
for Our app we have written detox e2e test cases,  want to run it on cloud devices like firebase test labs, browserstack , saucelabs...Does detox support run app on cloud devices?

### DISCUSSION THREAD

**Discussion by govindsain13** [wix/Detox#2952] [Q&A]
for Our app we have written detox e2e test cases,  want to run it on cloud devices like firebase test labs, browserstack , sauce labs. or any cloud devices service ..Does detox support run app on cloud devices?

**Comment by d4vidi** [comment]
None of those that you mentioned, unfortunately. We are close to maturing our support for `Genymotion`'s SaaS solution, if that helps: https://twitter.com/detoxe2e/status/1414140556238888961

For browserstack, there is a long and vague roadmap, but no guarantees at the moment.
